src/Runner/SshRunner.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Castor\Runner;
+
+use Spatie\Ssh\Ssh;
+use Symfony\Component\Process\Process;
+
+/** @internal */
+final class SshRunner
+{
+    public function __construct(
+        private ProcessRunner $processRunner,
+    ) {
+    }
+
+    public function execute(
+        string $command,
+        string $path,
+        string $host,
+        string $user,
+        array $sshOptions = [],
+        ?bool $quiet = null,
+        ?bool $allowFailure = null,
+        ?bool $notify = null,
+        ?float $timeout = null,
+    ): Process {
+        $ssh = $this->configuration($host, $user, $sshOptions);
+
+        if ($path) {
+            $command = sprintf('cd %s && %s', $path, $command);
+        }
+
+        return $this->run($ssh->getExecuteCommand($command), $quiet, $allowFailure, $notify, $timeout);
+    }
+
+    /**
+     * @param array{
+     *     'port'?: int,
+     *     'path_private_key'?: string,
+     *     'jump_host'?: string,
+     *     'multiplexing_control_path'?: string,
+     *     'multiplexing_control_persist'?: string,
+     *     'enable_strict_check'?: bool,
+     *     'password_authentication'?: bool,
+     * } $sshOptions
+     */
+    public function upload(
+        string $sourcePath,
+        string $destinationPath,
+        string $host,
+        string $user,
+        array $sshOptions = [],
+        ?bool $quiet = null,
+        ?bool $allowFailure = null,
+        ?bool $notify = null,
+        ?float $timeout = null,
+    ): Process {
+        $ssh = $this->configuration($host, $user, $sshOptions);
+
+        return $this->run($ssh->getUploadCommand($sourcePath, $destinationPath), $quiet, $allowFailure, $notify, $timeout);
+    }
+
+    /**
+     * @param array{
+     *     'port'?: int,
+     *     'path_private_key'?: string,
+     *     'jump_host'?: string,
+     *     'multiplexing_control_path'?: string,
+     *     'multiplexing_control_persist'?: string,
+     *     'enable_strict_check'?: bool,
+     *     'password_authentication'?: bool,
+     * } $sshOptions
```

Comment:
Could be a type in this class https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases

Reply:
Already changed :+1: